src/controllers/productController.ts

```diff
@@ -144,19 +144,45 @@ export async function getProducts(req: any, res: Response) {
 export class ProductController {
   static async getAvailableProduct(req: Request, res: Response) {
     try {
-      const allAvailableProducts = await db.Product.findAll({
-        where: {
-          isAvailable: true,
-        },
-      });
+      const page = parseInt(req.query.page as string, 10) || 1;
+      const productPerPage = 10;
+      const offset = (page - 1) * productPerPage;
+
+      const { count, rows: allAvailableProducts } =
+        await db.Product.findAndCountAll({
+          where: {
+            isAvailable: true,
+          },
+          limit: productPerPage,
+          offset: offset,
+        });
+
       if (!allAvailableProducts.length) {
         return res
           .status(404)
           .json({ message: 'No available products in our store' });
```

Comment:
If the product array is empty, it doesn't mean that it is an array. Consider returning an empty array